NaradaBotDemo/NaradaBotDemo/Source/ChatViewController.swift

```diff
@@ -117,23 +125,29 @@ class ChatViewController: JSQMessagesViewController, ApiAIChatDelegate, CardCell
         return
     }
     
+    //MARK: - Private API
+    
     //MARK: - ApiAIChatDelegate
     func addMessageFromApi(senderID: String, text: String) {
         self.addMessage(senderId: senderID, text: text)
         self.finishReceivingMessage()
     }
     
     func addCardFromApi(senderID: String, title: String, subtitle: String, image: String, action: String, buttonName: String) {
-        self.addCardMessage(senderId: senderId, title: title, subtitle: subtitle, image: image, action: action, buttonName: buttonName)
+        self.addCardMessage(senderId: senderID, title: title, subtitle: subtitle, image: image, action: action, buttonName: buttonName)
         self.finishReceivingMessage()
     }
     
     //MARK: - CardCellDelegate
-    func buttonPressed(action: String) {
+    func buttonPressed(title: String, subtitle: String, image: UIImage, action: String) {
```

Comment:
still i don't get why so many params everywhere. Do you need them all to discern what's clicked?